app/controllers/application_controller.rb

```diff
@@ -11,6 +11,16 @@ def current_user
   end
   helper_method :current_user
 
+  def logged_in?
+    session[:user_id].present?
+  end
+
+  def require_login
+    unless logged_in?
+      redirect_to new_session_path, alert: I18n.t("sessions.new.logged_in")
+    end
+  end
```

Comment:
These methods should be private so they could not be accessed from the outside world. The controller should look something like:

```
protect_from_forgery with: :exception

private

def current_user
end

def logged_in?
end

def require_login
end
```

Reply:
Done in sha: ae72639